.github/workflows/binary-publish.yml

```diff
@@ -0,0 +1,37 @@
+name: Publish binary
+on:
+  workflow_dispatch:
+
+jobs:
+  SetEnv:
+    environment: Testing
+    runs-on: ubuntu-latest
+    outputs:
+      version: ${{ steps.version.outputs.version }}
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Get version string
+        id: version
+        run: |
+          if ${{ startsWith(github.ref, 'refs/tags/') }}; then
+            echo "version=$(git describe --exact-match --tags $(git rev-parse HEAD))" | tee -a $GITHUB_OUTPUT
```

Comment:
YAML question - Do we typically not do any error handling? What's the behavior if say `git describe` fails?

Reply:
We don't really have much in terms of error handling in the CI. I might be useful to improve them, but don't think they are high up in the priority list.